show_client/arp_cli.go

```diff
@@ -52,38 +58,90 @@ func getArpTable(args sdc.CmdArgs, options sdc.OptionMap) ([]byte, error) {
 		}
 	}
 
-	output, err := common.GetDataFromHostCommand(cmd)
+	rawOutput, err := common.GetDataFromHostCommand(cmd)
 	if err != nil {
 		return nil, err
 	}
+	if strings.TrimSpace(rawOutput) == "" {
+		return []byte(`{"entries":[],"total_entries":0}`), nil
+	}
+	nbrdata := parseNbrData(rawOutput)
 
-	lines := strings.Split(output, "\n")
-	entries := []ArpEntry{}
-	for _, line := range lines {
-		line = strings.TrimSpace(line)
-		fields := strings.Fields(line)
+	bridgeMacList, err := common.FetchFDBData()
+	if err != nil {
+		return nil, err
+	}
+
+	entries := mergeNbrWithFDB(nbrdata, bridgeMacList)
 
-		// Skip headers, separators, summaries, or malformed lines
-		if len(fields) < OutputFieldsCount ||
-			strings.HasPrefix(fields[0], "-") ||
-			strings.Contains(line, "Address") ||
-			strings.Contains(line, "Total number of entries") {
+	sort.Slice(entries, func(i, j int) bool {
+		return entries[i].Address < entries[j].Address
+	})
+
+	response := ARPResponse{
+		Entries:         entries,
+		TotalEntryCount: len(entries),
+	}
+
+	if response.Entries == nil {
+		response.Entries = []ARPEntry{}
+	}
+
+	return json.Marshal(response)
+}
+
+func parseNbrData(output string) [][]string {
+	var nbrdata [][]string
+	for _, line := range strings.Split(output, "\n")[1:] {
+		if !strings.Contains(line, "ether") {
 			continue
 		}
-
-		for i := range fields {
-			fields[i] = strings.TrimSpace(fields[i])
+		var ent []string
+		fields := strings.Fields(line)
+		for i := 0; i < len(fields); i += 2 {
+			ent = append(ent, fields[i])
 		}
+		nbrdata = append(nbrdata, ent)
+	}
+	return nbrdata
+}
+
+func mergeNbrWithFDB(nbrdata [][]string, bridgeMacList []common.BridgeMacEntry) []ARPEntry {
+	var output []ARPEntry
+
+	for _, ent := range nbrdata {
+		vlan := "-"
+		if strings.Contains(ent[2], "Vlan") {
+			re := regexp.MustCompile(`\d+`)
```

Comment:
we can pull this out of the loop and specify vlan too: regexp.MustCompile(`^Vlan(\d+)$`)